codegen-renderers/src/main/kotlin/io/vrap/codegen/languages/java/commands/JavaCommandsRenderer.kt

```diff
@@ -0,0 +1,178 @@
+                |   @Override
+                |   public HttpRequestIntent httpRequestIntent() {
+                |      ${type.httpRequestIntentBody()}
+                |   }
+                |   
+                |   @Override
+                |   public ${type.javaReturnType()} deserialize(HttpResponse httpResponse) {
+                |       return SphereRequestUtils.deserialize(httpResponse, jacksonJavaType());
+                |   }   
+                |   
+                |   <${type.jacksonJavaTypeMethod()}>
+                |}
+                |
+                """.trimMargin().keepIndentation()
+        
+        return TemplateFile(
+            relativePath = "${vrapType.`package`}.${type.toRequestName()}".replace(".", "/") + ".java",
+            content = content
+        )
+    }
+
+    private fun Method.httpRequestIntentBody() : String? {
+        val path = this.resource().fullUri.template
+        val parts = path.split("/").filter { it.isNotEmpty() }
+        val pathArguments = parts.filter { it.startsWith("{") && it.endsWith("}")}
+        var stringFormat = path
+        
+        pathArguments.forEach { stringFormat = stringFormat.replace(it, "%s") }
+        
+        val stringFormatArgs = pathArguments
+                .map { it.replace("{", "").replace("}", "") }
+                .map { "this.$it" }
+                .joinToString(separator = ",")
+
+        val httpRequestIntentArguments = mutableListOf<String>()
+        httpRequestIntentArguments.add("HttpMethod.${this.method.name}")
+        
+        if(stringFormat.isNotEmpty() && stringFormatArgs.isNotEmpty()){
+            httpRequestIntentArguments.add("String.format(\"$stringFormat\", $stringFormatArgs)")
+        }
+        
+        if(this.bodyObjectName() != null){
+            httpRequestIntentArguments.add("SphereJsonUtils.toJsonString(${this.bodyObjectName()})")
+        }
+        
+        return "return HttpRequestIntent.of(${httpRequestIntentArguments.joinToString(separator = ", ")});"
+    }
+    
+    private fun Method.ofMethod(): String? {
+        val ofMethodArguments = mutableListOf<String>()
+        val constructorArguments = mutableListOf<String>()
+        
+        this.pathArguments().map { "final String $it" }.forEach { ofMethodArguments.add(it) }
+        constructorArguments.addAll(this.pathArguments())
+        
+        if(this.bodies != null && this.bodies.isNotEmpty()){
+            val methodBodyVrapType = this.bodies[0].type.toVrapType() as VrapObjectType
+            val methodBodyArgument = "final ${methodBodyVrapType.`package`}.${methodBodyVrapType.simpleClassName} ${methodBodyVrapType.simpleClassName.decapitalize()}"
+            ofMethodArguments.add(methodBodyArgument)
+            constructorArguments.add(methodBodyVrapType.simpleClassName.decapitalize())
+        }
+        
+        return """
+            |public static ${this.toRequestName()} of(${ofMethodArguments.joinToString(separator = ", ")}) {
```

Comment:
How about extracting `ofMethodArguments.joinToString(separator = ", ")` to a local var? This would make debugging a bit easier. WDYT?